staging/src/k8s.io/component-base/kflag/example/options/options.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"k8s.io/component-base/kflag"
+	"k8s.io/component-base/kflag/example/config"
+)
+
+// Flags that aren't part of the config
+type Flags struct {
+	OtherNum int32
+}
+
+func NewFlags() *Flags {
+	return &Flags{
+		OtherNum: 4,
+	}
+}
+
+func NewConfig() *config.Config {
+	c := &config.Config{}
+	config.Default(c)
+	return c
+}
+
+// We could have AddConfigFlags just return the function to apply
+// config flags, to keep each flag's registration and application
+// code close together:
+
+func AddConfigFlags(fs *kflag.FlagSet) (apply func(c *config.Config)) {
```

Comment:
Probably have a helper based on this inside the library itself? I can see many k8s components having to implement sort of the same thing here.

Reply:
Since this helper needs to understand the `Config` type, which kflag can't know about (it is different for any given component), I don't think we would be able to maintain compile-time type safety if we did that.

Reply:
It will be some boilerplate code to be implemented. But I think it is simple enough to be not an issue. And having type safety forces us to do that again and again unfortunately.

Reply:
Don't we all ;)